packages/promises/_test.pony

```diff
@@ -145,3 +149,150 @@ class iso _TestPromisesJoinThenReject is UnitTest
     a("a")
     b("b")
     c.reject()
+
+class iso _TestFlattenNextHappyPath is UnitTest
+  fun name(): String => "promises/Promise.flatten_next/happy_path"
+
+  fun apply(h: TestHelper) =>
+    let initial_string = "foo"
+    let second_string = "bar"
+
+    h.long_test(2_000_000_000)
+    h.expect_action(initial_string)
+    h.expect_action(second_string)
+
+    let start = Promise[String]
+    let inter = start.flatten_next[String](
+      _FlattenNextStartHelper~successful_fulfill(
+        second_string, h, initial_string),
+      _FlattenNextStartHelper~reject_fail_if_called(h)
+    )
+    inter.next[None](
+      _FlattenNextInterHelper~assert_equality(h, second_string),
+      _FlattenNextInterHelper~reject_fail_if_called(h)
+    )
+
+    start(initial_string)
+
+class iso _TestFlattenNextStartFulfillErrors is UnitTest
+  fun name(): String => "promises/Promise.flatten_next/start_fulfill_errors"
+  fun apply(h: TestHelper) =>
+    let initial_string = "foo"
+    let expected_reject_string = "rejected"
+
+    h.long_test(2_000_000_000)
+    h.expect_action(initial_string)
+    h.expect_action(expected_reject_string)
+
+    let start = Promise[String]
+    let inter = start.flatten_next[String](
+      _FlattenNextStartHelper~error_fulfill(h, initial_string),
```

Comment:
The error fulfill method up here seems to have a slight problem because this isn't the end of the test,
but error_fulfill marks complete(true)